contracts/Controller.sol

```diff
@@ -751,15 +761,24 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
     function _settleVault(Actions.SettleVaultArgs memory _args) internal onlyAuthorized(msg.sender, _args.owner) {
         require(_checkVaultId(_args.owner, _args.vaultId), "Controller: invalid vault id");
 
-        MarginVault.Vault memory vault = getVault(_args.owner, _args.vaultId);
-        bool hasShort = _isNotEmpty(vault.shortOtokens);
-        bool hasLong = _isNotEmpty(vault.longOtokens);
+        (MarginVault.Vault memory vault, uint256 typeVault, ) = getVault(_args.owner, _args.vaultId);
+
+        OtokenInterface otoken;
+
+        {
+            bool hasShort = _isNotEmpty(vault.shortOtokens);
+            bool hasLong = _isNotEmpty(vault.longOtokens);
+
+            require(hasShort || hasLong, "Controller: Can't settle vault with no otoken");
 
-        require(hasShort || hasLong, "Controller: Can't settle vault with no otoken");
+            otoken = hasShort ? OtokenInterface(vault.shortOtokens[0]) : OtokenInterface(vault.longOtokens[0]);
 
-        OtokenInterface otoken = hasShort
-            ? OtokenInterface(vault.shortOtokens[0])
-            : OtokenInterface(vault.longOtokens[0]);
+            if (hasLong) {
+                OtokenInterface longOtoken = OtokenInterface(vault.longOtokens[0]);
+
+                longOtoken.burnOtoken(address(pool), vault.longAmounts[0]);
+            }
+        }
```

Comment:
why do we need this {} block?

Reply:
stack too deep error 😅

Reply:
That is basically a diff scope to declare variables in, in this case the `hasShort` and `hasLong` will not exist outside of that block.

Reply:
wow this is a cool hack!

Reply:
Will un-resolve this for now so that other can see this...

Reply:
This is cool! :D